### PR TITLE
Migrate jdbcCardDao to spring:

### PR DESCRIPTION
**Pull request by RomanKlishch**
- changed methods in class JdbcCardDao;
- changed classes CardRowMapper and CardAndCongratulationRowMapper;
- removed classes JdbcConfigureFactory, PropertyReader and test for them;
- added new tests

### CODE REVIEW

File: src/main/java/com/greetingcard/web/servlet/card/GetCardsServlet.java [hunk 1]
```diff
@@ -16,7 +16,8 @@
 
 @Slf4j
 public class GetCardsServlet extends HttpServlet {
-    private CardService cardService = ServiceLocator.getBean("DefaultCardService");
+    @Autowired
+    private CardService cardService;
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (AlexKievDeveloper) [inline src/main/java/com/greetingcard/web/servlet/card/GetCardsServlet.java:23]: Why didn't you delete this servlet?

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java [hunk 2]
```diff
@@ -22,7 +21,6 @@
 import java.util.Objects;
 
 @Slf4j
-@Setter
```
Comment (AlexKievDeveloper) [inline src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java:25]: Why you decide to delete my annotation?

**Review by AlexKievDeveloper — APPROVED** [review]

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java [hunk 3]
```diff
@@ -9,160 +8,102 @@
...
+    @Autowired
+    public JdbcCardDao(CongratulationDao congratulationDao, JdbcTemplate jdbcTemplate, NamedParameterJdbcTemplate namedParameterJdbcTemplate, TransactionTemplate transactionTemplate) {
+        this.congratulationDao = congratulationDao;
+        this.jdbcTemplate = jdbcTemplate;
+        this.namedParameterJdbcTemplate = namedParameterJdbcTemplate;
+        this.transactionTemplate = transactionTemplate;
     }
 
     @Override
     public List<Card> getAllCardsByUserId(long id) {
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java:51]: Why do we to name parameter id and in next method - userId

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java [hunk 4]
```diff
@@ -9,160 +8,102 @@
...
-                    cardsList.add(card);
-                }
-            }
-            return cardsList;
-        } catch (SQLException e) {
-            log.error("Exception while getting my cards from DB by user id: {} and role id: {}", userId, roleId, e);
-            throw new RuntimeException("Exception while getting my cards from DB by user id: " + userId +
-                    " and role id: " + roleId, e);
-        }
+        MapSqlParameterSource namedParameters = new MapSqlParameterSource("userId", userId);
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java:58]: Why do we use MapSqlParameterSource and in before method - SqlParameterSource

File: src/main/java/com/greetingcard/security/DefaultSecurityService.java [hunk 5]
```diff
@@ -17,13 +16,13 @@
...
 public class DefaultSecurityService implements SecurityService {
-
-//    private PropertyReader propertyReader = ServiceLocator.getBean("PropertyReader");
-    @Autowired
-    private PropertyReader propertyReader;
-    @Autowired
+    @Value("${algorithm}")
+    private String algorithm;
+    @Value("${iteration}")
+    private int iteration;
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/security/DefaultSecurityService.java:22]: Have you tested this? Is it work?

File: src/main/java/com/greetingcard/web/servlet/card/GetCardsServlet.java (hunk 1 above)
Comment (OleksandrGubariev) [reply src/main/java/com/greetingcard/web/servlet/card/GetCardsServlet.java:23]: ++

File: src/main/java/com/greetingcard/web/servlet/congratulation/CongratulationServlet.java [hunk 6]
```diff
@@ -18,7 +18,8 @@
 
 @Slf4j
 public class CongratulationServlet extends HttpServlet {
-    private CongratulationService congratulationService = ServiceLocator.getBean("DefaultCongratulationService");
+    @Autowired
+    private CongratulationService congratulationService;
 
     @Override
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/web/servlet/congratulation/CongratulationServlet.java:25]: Why didn't you remove this servlet?

File: src/main/java/com/greetingcard/web/servlet/user/RegistrationServlet.java [hunk 7]
```diff
@@ -17,7 +17,8 @@
 
 @Slf4j
 public class RegistrationServlet extends HttpServlet {
-    private SecurityService securityService = ServiceLocator.getBean("DefaultSecurityService");
+    @Autowired
+    private SecurityService securityService;
 
     @Override
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/web/servlet/user/RegistrationServlet.java:24]: Why didn't you remove this servlet?

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java [hunk 8]
```diff
@@ -9,160 +8,102 @@
...
-    private final CongratulationDao congratulationDao = ServiceLocator.getBean("JdbcCongratulationDao");
-    private final DataSource dataSource;
+    private CongratulationDao congratulationDao;
+    private JdbcTemplate jdbcTemplate;
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+    private TransactionTemplate transactionTemplate;
 
-    public JdbcCardDao(DataSource dataSource) {
-        this.dataSource = dataSource;
+    @Autowired
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java:42]: Why do you add @Autowired?

File: src/main/java/com/greetingcard/web/servlet/card/GetCardsServlet.java (hunk 1 above)
Comment (RomanKlishch) [reply src/main/java/com/greetingcard/web/servlet/card/GetCardsServlet.java:23]: I removed ServiceLocator

**Review by OleksandrGubariev — APPROVED** [review]

**Pull request merged**